HW4_Mukhametshina/amino_acid_tools.py

```diff
@@ -0,0 +1,241 @@
+             'Ala', 'Arg', 'Asn', 'Asp', 'Cys', 'Glu', 'Gln', 'Gly', 'His', 'Ile', 'Leu', 'Lys', 'Met', 'Phe', 'Pro',
+             'Ser', 'Thr', 'Trp', 'Tyr', 'Val', 'Sec', 'Pyl']
+codon_table = {
+    'A': ['GCU', 'GCC', 'GCA', 'GCG'],
+    'R': ['CGU', 'CGC', 'CGA', 'CGG', 'AGA', 'AGG'],
+    'N': ['AAU', 'AAC'],
+    'D': ['GAU', 'GAC'],
+    'C': ['UGU', 'UGC'],
+    'Q': ['CAA', 'CAG'],
+    'E': ['GAA', 'GAG'],
+    'G': ['GGU', 'GGC', 'GGA', 'GGG'],
+    'H': ['CAU', 'CAC'],
+    'I': ['AUU', 'AUC', 'AUA'],
+    'L': ['UUA', 'UUG', 'CUU', 'CUC', 'CUA', 'CUG'],
+    'K': ['AAA', 'AAG'],
+    'M': ['AUG'],
+    'F': ['UUU', 'UUC'],
+    'P': ['CCU', 'CCC', 'CCA', 'CCG'],
+    'S': ['UCU', 'UCC', 'UCA', 'UCG', 'AGU', 'AGC'],
+    'T': ['ACU', 'ACC', 'ACA', 'ACG'],
+    'W': ['UGG'],
+    'Y': ['UAU', 'UAC'],
+    'V': ['GUU', 'GUC', 'GUA', 'GUG'],
+    'STOP': ['UAA', 'UAG', 'UGA'],
+    'f': ['uuu', 'uuc'],
+    'l': ['uua', 'uug', 'cuu', 'cuc', 'cua', 'cug'],
+    's': ['ucu', 'ucc', 'uca', 'ucg', 'agu', 'agc'],
+    'y': ['uau', 'uac'],
+    'c': ['ugu', 'ugc'],
+    'w': ['ugg'],
+    'p': ['ccu', 'ccc', 'cca', 'ccg'],
+    'h': ['cau', 'cac'],
+    'q': ['caa', 'cag'],
+    'r': ['cgu', 'cgc', 'cga', 'cgg', 'aga', 'agg'],
+    'i': ['auu', 'auc', 'aua'],
+    'm': ['aug'],
+    't': ['acu', 'acc', 'aca', 'acg'],
+    'n': ['aau', 'aac'],
+    'k': ['aaa', 'aag'],
+    'v': ['guu', 'guc', 'gua', 'gug'],
+    'a': ['gcu', 'gcc', 'gca', 'gcg'],
+    'd': ['gau', 'gac'],
+    'e': ['gaa', 'gag'],
+    'g': ['ggu', 'ggc', 'gga', 'ggg'],
+    'stop': ['uaa', 'uag', 'uga']
+}
+weight_amino = [71.08, 156.2, 114.1, 115.1, 103.1, 129.1, 128.1, 57.05, 137.1, 113.2, 113.2, 128.2, 131.2, 147.2, 97.12, 87.08,
+         101.1, 186.2, 163.2, 99.13, 168.05, 255.3,
+         71.08, 156.2, 114.1, 115.1, 103.1, 129.1, 128.1, 57.05, 137.1, 113.2, 113.2, 128.2, 131.2, 147.2, 97.12, 87.08,
+         101.1, 186.2, 163.2, 99.13, 168.05, 255.3]
+
+import random
+
+def long_amino_code(sequence):
+    """
+        Function translates a given sequence of one-letter amino acids
+    into a more understandable sequence of amino acids consisting of three letters
+
+        Parameters:
+            sequence (str): each letter refers to one-letter coded proteinogenic amino acids or "random"
+        Returns:
+            (str) translated in three-letter code
+    """
+    if sequence != 'random':
+        d_names = dict(zip(short_code, long_code))
+        recording = sequence.maketrans(d_names)
+        return sequence.translate(recording)
+    else:
+        len = int(input("введите желаемую длину: "))
+        bases = list(amino_acid)
+        amino_sequencqe = ''.join(random.choice(bases) for i in range(len))
+        d_names = dict(zip(short_code, long_code))
+        recording = amino_sequencqe.maketrans(d_names)
+        return "рандомная последовательнсть", amino_sequencqe, amino_sequencqe.translate(recording)
+
+def molecular_weight(sequence):
```

Comment:
```suggestion
def calc_molecular_weight(sequence):
```
Глагол. Но можно сокращения, если они понятные и общепринятые